extractData.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"os"
+)
+
+func extractData[T Wind | Basic](w http.ResponseWriter, filenm string, params T) error {
+	// Reading data frm the file
+	jsonData, err := os.ReadFile(filenm)
+	if err != nil {
+		log.Println("Couldn't read from file")
+		return err
+	}
+
+	// Extracting what I wanna send to the user
+	err = json.Unmarshal(jsonData, &params)
+	if err != nil {
+		log.Println("Couldn't unMarshall")
+		return err
+	}
+
+	// converting the extracted data to json again
+	dat, err := json.Marshal(params)
+	if err != nil {
+		log.Printf("Failed to marshal JSON response: %v", params)
+		return err
+	}
+
+	// Sending the Response
+	w.Header().Add("Content-Type", "application/json")
```

Comment:
We must not write to the request within a function that is only meant to extract data. A function does one thing and that thing well. It need not receive the request response parameter. And, it shouldn't write to it because that's not expected.

Someother function would want to extract data but it may not intend to write to any http request at all.

Reply:
I see. Will do it.